### PR TITLE
fix: add transaction simulation polling

### PR DESCRIPTION
**Pull request by dhruvkelawala**
- also handle loading states

### Issue / feature description

Add polling on Transaction Simulation to always get the latest data. Also added better loading state handling

### Checklist

- [x] Rebased to the last commit of the target branch (or merged)
- [x] Code self-reviewed
- [x] Code self-tested
- [x] Tests updated (if needed)
- [x] All tests are passing locally

### CODE REVIEW

**Review by janek26 — APPROVED** [review]

File: packages/extension/src/ui/features/actions/transaction/ApproveTransactionScreen/DappHeader/TransactionIcon/NftTransactionIcon.tsx [hunk 1]
```diff
@@ -16,25 +16,28 @@ export const NftTransactionIcon: FC<NFTPictureProps> = ({
...
+    if (isValidating && !error) {
+      return (
+        <IconWrapper>
+          <Skeleton
+            borderRadius="2xl"
+            startColor="neutrals.500"
+            endColor="neutrals.700"
+          />
+        </IconWrapper>
+      )
```
Comment (diegodelrieu) [inline packages/extension/src/ui/features/actions/transaction/ApproveTransactionScreen/DappHeader/TransactionIcon/NftTransactionIcon.tsx:39]: Isnt there a case that is not covered here ?

File: packages/extension/src/ui/features/actions/transaction/useTransactionSimulation.ts [hunk 2]
```diff
@@ -56,6 +56,7 @@ export const useTransactionSimulation = ({
     transactionSimulationFetcher,
     {
       revalidateOnFocus: false,
+      refreshInterval: 15e3,
```
Comment (simonheys) [inline packages/extension/src/ui/features/actions/transaction/useTransactionSimulation.ts:59]: Just to sense-check - will it throttle now on Testnet if BE repeatedly returns 500?

Comment (dhruvkelawala) [reply packages/extension/src/ui/features/actions/transaction/useTransactionSimulation.ts:59]: I don't think so it throttles if it's on 15s interval.

File: packages/extension/src/ui/features/actions/transaction/ApproveTransactionScreen/DappHeader/TransactionIcon/NftTransactionIcon.tsx (hunk 1 above)
Comment (dhruvkelawala) [reply packages/extension/src/ui/features/actions/transaction/ApproveTransactionScreen/DappHeader/TransactionIcon/NftTransactionIcon.tsx:43]: Which case?

**Review by simonheys — APPROVED** [review]

File: packages/extension/src/ui/features/actions/transaction/ApproveTransactionScreen/DappHeader/TransactionIcon/NftTransactionIcon.tsx (hunk 1 above)
Comment (diegodelrieu) [reply packages/extension/src/ui/features/actions/transaction/ApproveTransactionScreen/DappHeader/TransactionIcon/NftTransactionIcon.tsx:43]: ah nvm all good

**Review by diegodelrieu — APPROVED** [review]

**Pull request merged**